nifi-nar-bundles/nifi-influxdb-bundle/nifi-influxdb-processors/src/main/java/org/apache/nifi/processors/influxdb/PutInfluxDB.java

```diff
@@ -0,0 +1,176 @@
+
+    public static final PropertyDescriptor CONSISTENCY_LEVEL = new PropertyDescriptor.Builder()
+            .name("influxdb-consistency-level")
+            .displayName("Consistency Level")
+            .description("InfluxDB consistency level")
+            .required(true)
+            .defaultValue(CONSISTENCY_LEVEL_ONE.getValue())
+            .expressionLanguageSupported(true)
+            .allowableValues(CONSISTENCY_LEVEL_ONE, CONSISTENCY_LEVEL_ANY, CONSISTENCY_LEVEL_ALL, CONSISTENCY_LEVEL_QUORUM)
+            .build();
+
+    public static final PropertyDescriptor RETENTION_POLICY = new PropertyDescriptor.Builder()
+            .name("influxdb-retention-policy")
+            .displayName("Retention Policy")
+            .description("Retention policy for the saving the records")
+            .defaultValue("autogen")
+            .required(true)
+            .expressionLanguageSupported(true)
+            .addValidator(StandardValidators.NON_EMPTY_VALIDATOR)
+            .build();
+
+    private static final Set<Relationship> relationships;
+    private static final List<PropertyDescriptor> propertyDescriptors;
+
+    static {
+        final Set<Relationship> tempRelationships = new HashSet<>();
+        tempRelationships.add(REL_SUCCESS);
+        tempRelationships.add(REL_FAILURE);
+        relationships = Collections.unmodifiableSet(tempRelationships);
+
+        final List<PropertyDescriptor> tempDescriptors = new ArrayList<>();
+        tempDescriptors.add(DB_NAME);
+        tempDescriptors.add(INFLUX_DB_URL);
+        tempDescriptors.add(USERNAME);
+        tempDescriptors.add(PASSWORD);
+        tempDescriptors.add(CHARSET);
+        tempDescriptors.add(CONSISTENCY_LEVEL);
+        tempDescriptors.add(RETENTION_POLICY);
+        tempDescriptors.add(MAX_RECORDS_SIZE);
+        propertyDescriptors = Collections.unmodifiableList(tempDescriptors);
+    }
+
+    @Override
+    public Set<Relationship> getRelationships() {
+        return relationships;
+    }
+
+    @Override
+    public final List<PropertyDescriptor> getSupportedPropertyDescriptors() {
+        return propertyDescriptors;
+    }
+
+    @OnScheduled
+    public void onScheduled(final ProcessContext context) {
+        super.onScheduled(context);
+    }
+
+    @Override
+    public void onTrigger(final ProcessContext context, final ProcessSession session) throws ProcessException {
+        FlowFile flowFile = session.get();
+        if (flowFile == null) {
+            return;
+        }
+
+        if ( flowFile.getSize() == 0) {
+            getLogger().error("Empty measurements");
+            flowFile = session.putAttribute(flowFile, INFLUX_DB_ERROR_MESSAGE, "Empty measurement size " + flowFile.getSize());
+            session.transfer(flowFile, REL_FAILURE);
+            return;
+        }
+
+        if ( flowFile.getSize() > maxRecordsSize) {
```

Comment:
This seems a bit stern, what would a user do if the size of the records is too large? Can it be split using NiFi processors? If not, it seems like the data would have no place to go.

Reply:
@mans2singh Correct me if I am wrong, but I think I saw something in the documentation that said that InfluxDB's REST handler only accepts a few MB of data/request (5MB/request?). Is that right?

Reply:
I could not find any documentation in the InfluxDB docs.